native/packages/react-native-bpk-component-button/src/BpkButton-test.common.js

```diff
@@ -16,155 +16,173 @@
       ).toJSON();
-      expect(tree).toMatchSnapshot();
-    });
+       expect(tree).toMatchSnapshot();
+     });
 
-    it('should support the "iconOnly" and "large" property', () => {
-      const tree = renderer.create(
-        <BpkButton iconOnly large icon title="Lorem ipsum" onPress={onPressFn} />,
+     it('should support the "iconOnly" and "large" property', () => {
+       const tree = renderer.create(
+         <BpkButton iconOnly large icon="baggage" title="Lorem ipsum" onPress={onPressFn} />,
       ).toJSON();
-      expect(tree).toMatchSnapshot();
-    });
-    it('should support the "icon" and "large" property', () => {
-      const tree = renderer.create(
-        <BpkButton icon large title="Lorem ipsum" onPress={onPressFn} />,
+       expect(tree).toMatchSnapshot();
+     });
+     it('should support the "icon" and "large" property', () => {
+       const tree = renderer.create(
+         <BpkButton icon="baggage" large title="Lorem ipsum" onPress={onPressFn} />,
       ).toJSON();
-      expect(tree).toMatchSnapshot();
-    });
+       expect(tree).toMatchSnapshot();
+     });
 
-    it('should support the "disabled" property', () => {
-      const tree = renderer.create(
-        <BpkButton disabled title="Lorem ipsum" onPress={onPressFn} />,
+     it('should support the "disabled" property', () => {
+       const tree = renderer.create(
+         <BpkButton disabled title="Lorem ipsum" onPress={onPressFn} />,
+      ).toJSON();
+       expect(tree).toMatchSnapshot();
+     });
+
+     it('should support having an icon as well as a title', () => {
+       const tree = renderer.create(
+         <BpkButton
+           icon="baggage"
+           title="Lorem ipsum"
+           onPress={onPressFn}
+         />,
       ).toJSON();
-      expect(tree).toMatchSnapshot();
-    });
-
-    it('should support having an icon as well as a title', () => {
-      const tree = renderer.create(
-        <BpkButton
-          icon={<Image source="../rightarrow_360.png" />}
-          title="Lorem ipsum"
-          onPress={onPressFn}
-        />,
+       expect(tree).toMatchSnapshot();
+     });
+     it('should support having only an icon', () => {
+       const tree = renderer.create(
+         <BpkButton
+           title="Lorem ipsum"
+           icon="baggage"
+           iconOnly
+           onPress={onPressFn}
+         />,
       ).toJSON();
-      expect(tree).toMatchSnapshot();
-    });
-    it('should support having only an icon', () => {
-      const tree = renderer.create(
-        <BpkButton
-          title="Lorem ipsum"
-          icon={<Image source="../rightarrow_360.png" />}
-          iconOnly
-          onPress={onPressFn}
-        />,
+       expect(tree).toMatchSnapshot();
+     });
+
+     it('should support elements as icons', () => {
+       const tree = renderer.create(
+         <BpkButton
+           title="Lorem ipsum"
+           icon={<Text>foo</Text>}
```

Comment:
do you think is a good idea to add in the test a behaviour that we don't want to support? I'd use BpkIcon as element here, for people like me that look at the test to see how to use something, it'll encourage to use BpkIcon

Reply:
I see what you're saying, but then also, we do technically *support* it, we just don't *recommend* it. I'd want to know if a change we made stopped this working correctly, for example.

For me, it's the same as how we have tests for having both text and an icon in the button, even though it's not something we display in the docs.

However if we want to steer users away from this edge case it should be removed from the code sample in `readme.md`, I think.